pkg/athena/driver/connection.go

```diff
@@ -20,26 +24,53 @@ func newConnection(api *api.API) *conn {
 	}
 }
 
-func (c *conn) QueryContext(ctx context.Context, query string, _ []driver.NamedValue) (driver.Rows, error) {
+func (c *conn) StartQuery(ctx context.Context, query string, args ...interface{}) (string, error) {
 	output, err := c.api.Execute(ctx, &sqlAPI.ExecuteQueryInput{Query: query})
 	if err != nil {
-		return nil, err
+		return "", err
 	}
+	return output.ID, nil
+}
 
-	if err := sqlAPI.WaitOnQuery(ctx, c.api, output); err != nil {
-		return nil, err
+func (c *conn) GetQueryID(ctx context.Context, query string, args ...interface{}) (bool, string, error) {
+	return c.api.GetQueryID(ctx, query, args)
+}
+
+func (c *conn) QueryStatus(ctx context.Context, queryID string) (awsds.QueryStatus, error) {
+	status, err := c.api.Status(ctx, &sqlAPI.ExecuteQueryOutput{ID: queryID})
+	if err != nil {
+		return awsds.QueryUnknown, err
 	}
+	var returnStatus awsds.QueryStatus
+	switch status.State {
+	case athena.QueryExecutionStateQueued:
+		returnStatus = awsds.QuerySubmitted
+	case athena.QueryExecutionStateRunning:
+		returnStatus = awsds.QueryRunning
+	case athena.QueryExecutionStateSucceeded:
+		returnStatus = awsds.QueryFinished
+	case athena.QueryExecutionStateCancelled:
+		returnStatus = awsds.QueryCanceled
+	case athena.QueryExecutionStateFailed:
+		returnStatus = awsds.QueryFailed
```

Comment:
something about this switchcase is so satisfying to me lol. Good work y'all. Maybe worth logging a debug line or something in case people are curious?